sdlib/d/gc/extent.d

```diff
@@ -334,8 +342,7 @@ public:
 		}
 
 		// Encode freespace and write it to the last byte (or two bytes) of alloc.
-		writePackedFreeSpace(cast(ushort*) (address + slotSize - 2),
-		                     freeSpace & ushort.max);
+		writePackedFreeSpace(cast(ushort*) slotEndPtr(index) - 2, freeSpace & ushort.max);
```

Comment:
You are repeating the cast and the minus 2 now. I don't get it, there was a function with two call site and a function with one and you inline the function with 2 call site. This makes no sense.